spec/service/integration/person_intergration_spec.rb

```diff
@@ -0,0 +1,337 @@
+    end
+
+
+
+  end
+
+  context "#update" do
+
+    context "Person exists" do
+      let(:person){FactoryBot.create(:person)} # mock an instance of a person for testing
+
+      it "successful update http 200" do
+        # Add the mock person ID to the query
+        patch "/1/Person/#{person.id}",
+        {"name": "John"}.to_json,
+        { 'CONTENT_TYPE' => 'application/json; charset=utf-8' }
+        expect(last_response.status).to eq 200
+      end
+
+      it "updates the mock persons name" do
+        # Add the mock person ID to the query
+        patch "/1/Person/#{person.id}",
+        {"name": "John", "date_of_birth": "1996-01-01"}.to_json,
+        { 'CONTENT_TYPE' => 'application/json; charset=utf-8' }
+        message =%({"name": "John" , "date_of_birth": "1996-01-01" , "kind": "Person"})
+        expect(last_response.body).to be_json_eql(message)
+      end
+
+      it "returns 422 when date of birth is invalid" do
+        # Add the mock person ID to the query
+        patch "/1/Person/#{person.id}",
+        {"name": "John", "birthday": "today"}.to_json,
+        { 'CONTENT_TYPE' => 'application/json; charset=utf-8' }
+        expect(last_response.status).to eq 422
+      end
+    end
+
+    context "Person doesn't exist" do
+      let(:person){FactoryBot.create(:person)} # mock an instance of a person for testing
+      let(:id) { Hoodoo::UUID.generate}# generate a random UUID to mock an id that is not in the db.
+
+      it "returns 404 when updating an invalid id" do
+        patch "/1/Person/#{id}",
+        {"name": "John"}.to_json,
+        { 'CONTENT_TYPE' => 'application/json; charset=utf-8' }
+        expect(last_response.status).to eq 404
+      end
+
+      it "returns the correct error message when attempting to update an invalid id" do
+        patch "/1/Person/#{id}",
+        {"name": "John"}.to_json,
+        { 'CONTENT_TYPE' => 'application/json; charset=utf-8' }
+        message = %({
+          "errors": [
+            {
+              "code": "generic.not_found",
+              "message": "Resource not found",
+              "reference": "#{id}"
+            }
+          ],
+          "kind": "Errors"})
+        expect(last_response.body).to be_json_eql(message).excluding("interaction_id")
+      end
+
+      it "returns the correct error message when attempting to update an invalid id" do
+        patch "/1/Person/#{person.id}",
+        {"name": "John", "number_of_cats": "12"}.to_json,
+        { 'CONTENT_TYPE' => 'application/json; charset=utf-8' }
+        message = %({
+          "errors": [
+            {
+              "code": "generic.invalid_parameters",
+              "message": "Body data contains unrecognised or prohibited fields",
+              "reference": "number_of_cats"
+            }
+          ],
+          "kind": "Errors"})
+        expect(last_response.body).to be_json_eql(message).excluding("interaction_id")
+      end
```

Comment:
The test doesn't match the comment in the 'it' statement.  

Maybe we don't need this test. It looks like the one earlier on line 70 `it "renders incorrectly when an addition field is present."`